hazelcast/src/main/java/com/hazelcast/client/impl/spi/ProxyManager.java

```diff
@@ -462,6 +456,35 @@ void set(Object o) {
                 notifyAll();
             }
         }
+
+        @Override
+        public boolean block() throws InterruptedException {
+            if (Thread.currentThread().isInterrupted()
+                    || isReleasable()) {
+                return true;
+            }
+
+            boolean interrupted = false;
+            synchronized (this) {
+                while (proxy == null) {
+                    try {
+                        wait();
+                    } catch (InterruptedException e) {
```

Comment:
I think we don't need to catch `InterruptedException` , just let it be thrown.

Reply:
removed catch block

Reply:
As discussed on slack, removing the `catch` block would change code execution path, so it was restored.